### DISCUSSION TITLE
TUDataset() "PROTEINS" Parameters

### DISCUSSION THREAD

**Discussion by maes21353** [pyg-team/pytorch_geometric#6988] [General]
Hi, 

I am currently working with the PROTEINS dataset from TUDataset(). The way I am loading my data is as follows:

dataset = TUDataset(root='.', name='PROTEINS', use_node_attr = True).shuffle()

Now, when I print the number of features, i.e., dataset.num_features I am getting 4 although the dataset says there are 3 features. 

Also printing the number of nodes keeps giving me a different number with every run. I am doing it as follows:

print('Number of nodes:', dataset[0].x.shape[0])

Why am I encountering such behavior?  Can someone tell me what dataset[0].x is?

**Comment by EdisonLeeeee** [comment]
1. You are using `use_node_attr=True` to add additional continuous node attributes (1-dimensional)
2. The dataset is shuffled each time by the `.shuffle()` call, so the number of nodes in each graph may vary between runs.
